pyfr/plugins/tavg.py

```diff
@@ -118,23 +126,11 @@ def _eval_acc_exprs(self, intg):
             # Prepare the substitutions dictionary
             subs = dict(zip(pnames, psolns))
 
-            # Compute any required gradients
+            # Prepare any required gradient
             if self._gradpnames:
-                # Gradient operator and J^-T matrix
-                gradop, rcpjact = self._gradop[i], self._rcpjact[i]
-                nupts = gradop.shape[1]
-
-                for pname in self._gradpnames:
-                    psoln = subs[pname]
-
-                    # Compute the transformed gradient
-                    tgradpn = gradop @ psoln
-                    tgradpn = tgradpn.reshape(self.ndims, nupts, -1)
-
-                    # Untransform this to get the physical gradient
-                    gradpn = np.einsum('ijkl,jkl->ikl', rcpjact, tgradpn)
-                    gradpn = gradpn.reshape(self.ndims, nupts, -1)
-
+                gradps = grads_eles[j]
```

Comment:
See proposed simplification for the integrate plugin.

Reply:
Done